hashtypechecker.cpp

```diff
@@ -0,0 +1,99 @@
+#include "hashtypechecker.h"
+
+#include <QtDebug>
+
+HashTypeChecker::HashTypeChecker()
+{
+    connect(&m_john,SIGNAL(finished(int, QProcess::ExitStatus)),this,SLOT(startParsing()));
+    connect (&m_john, SIGNAL(readyReadStandardOutput()), this, SLOT(processOutput())); // connect process signals with your code
+}
+
+HashTypeChecker::~HashTypeChecker()
+{
+
+}
+void HashTypeChecker::start(QString& pathToJohn, QString& pathToPwdFile)
+{
+    m_john.start(pathToJohn + " --show=types " + pathToPwdFile);
+}
+void HashTypeChecker::terminate()
+{
+    if (m_john.state() != QProcess::NotRunning)
+        m_john.terminate();
+    m_john.waitForFinished(500);
+    if (m_john.state() != QProcess::NotRunning)
+        m_john.kill();
+
+    // Process what have been done so far from JohnOutput ...
+    startParsing();
+}
+void HashTypeChecker::startParsing()
+{
+    QtConcurrent::run(this,&HashTypeChecker::parseJohnAnswer);
+}
+
+void HashTypeChecker::processOutput()
+{
+  m_johnOutput.append(m_john.readAllStandardOutput()); // read normal output
+
+}
+void HashTypeChecker::parseJohnAnswer()
```

Comment:
@AlekseyCherepanov @shinnok 
I implemented a parser which look at all fields from --show=types and saves them in case we decide to use them later. However, to save CPU/memory, since we only want the LABEL, I could make a new method "getTypeOnly" which would skip unwanted fields(currentIndex+=7) for example and not save them in memory. In fact, until we use other fields ,I could make a light method which doesn't use Hash and HashFormat structs and only a std::vector<QStringList> to populate m_data in the filetablemodel. I'd like your opinion about this thanks guys